client/securedrop_client/config.py

```diff
@@ -37,28 +38,45 @@ class Config:
     mapping = {
         "gpg_domain": "QUBES_GPG_DOMAIN",
         "journalist_key_fingerprint": "SD_SUBMISSION_KEY_FPR",
+        "download_retry_limit": "SD_DOWNLOAD_RETRY_LIMIT",
     }
 
-    gpg_domain: str
     journalist_key_fingerprint: str
+    gpg_domain: str | None = None
+    download_retry_limit: int = 3
 
     @classmethod
-    def load(self) -> "Config":
+    def load(cls) -> "Config":
         """For each attribute, look it up from either QubesDB or the environment."""
         config = {}
 
         with try_qubesdb() as db:
-            for store, lookup in self.mapping.items():
+            for field in fields(cls):
+                lookup = cls.mapping[field.name]
                 if db:
                     logger.debug(f"Reading {lookup} from QubesDB")
                     value = db.read(f"/vm-config/{lookup}")
+                    source = "QubesDB"
                     if not value or len(value) == 0:
-                        raise KeyError(f"Could not read from QubesDB: {lookup}")
-
+                        # Normalize
+                        value = None
                 else:
                     logger.debug(f"Reading {lookup} from environment")
                     value = os.environ.get(lookup)
```

Comment:
Also for parity, do we care about the case where `len(os.environ.get(lookup)) == 1`?

Reply:
I assume you meant 0, and yes, good point. This was silently being handled by the `if not value` block below, let's handle it explicitly. Done.